### PR TITLE
Fix PV validation

### PR DESCRIPTION
**Pull request by nirs**
When comparing PVs, skip comparing unset optional fields. This breaks validation when using
KubeVirt VM, and actual resources in the system do not match the backed up resources in
the s3 store, differing only in the optional "kind" field.

I could not reproduce the mismatch between s3 and actual resources, both
match after deploy and failover. So I would say this issue is not reproducible.

But after creating a mismatch manually by editing the PV claimRef, the issue
was reproduced, and replacing ramen image with the fix solved the issue.

For testing use: quay.io/nirsof/ramen-operator:validate-pvc-v1

Bug: https://bugzilla.redhat.com/2262455

### CODE REVIEW

**Review by ELENAGER — APPROVED** [review]

File: controllers/util/misc.go [hunk 1]
```diff
@@ -115,3 +115,8 @@ func UpdateStringMap(dst *map[string]string, src map[string]string) {
 		(*dst)[key] = val
 	}
 }
+
+// OptionalEqual returns True if optinal field values are qual, or one of them is unset.
```
Comment (BenamarMk) [inline controllers/util/misc.go:119]: (nit) equal is mispelled --> `values are qual`...

**Review by BenamarMk — APPROVED** [review]

File: controllers/util/misc.go (hunk 1 above)
Comment (netzzer) [reply controllers/util/misc.go:119]: optinal -> optional (typo)

Comment (nirs) [reply controllers/util/misc.go]: Fixed

File: controllers/util/misc.go [hunk 2]
```diff
@@ -115,3 +115,8 @@ func UpdateStringMap(dst *map[string]string, src map[string]string) {
 		(*dst)[key] = val
 	}
 }
+
+// OptionalEqual returns True if optinal field values are equal, or one of them is unset.
```
Comment (nirs) [inline controllers/util/misc.go:119]: optinal -> optional

Comment (nirs) [reply controllers/util/misc.go]: fixed

**Review by ShyamsundarR — APPROVED** [review]
@nirs Any reason this is still in draft?

**Comment by nirs** [comment]
> @nirs Any reason this is still in draft?

Change was not tested yet.

**Pull request merged**